framework/src/modules/token/commands/transfer.ts

```diff
@@ -43,6 +48,42 @@ export class TransferCommand extends BaseCommand {
 
 		try {
 			validator.validate(transferParamsSchema, params);
+
+			const senderAddress = utils.hash(context.transaction.senderAddress);
+			const tokenID = context.transaction.senderAddress;
+			const userStore = this.stores.get(UserStore);
+
+			const balanceChecks: { [key: string]: bigint } = {};
+
+			if (userStore.getKey(params.recipientAddress, tokenID).length === 0) {
+				if (params.accountInitializationFee !== USER_SUBSTORE_INITIALIZATION_FEE) {
+					throw new Error('Invalid account initialization fee');
+				}
+
+				balanceChecks[
+					TOKEN_ID_STORE_INITIALIZATION.toString('hex')
```

Comment:
We need to compute the mainchain token ID from it's own chain ID here because in LIP it mentiones the LSK token ID, but for testnet mainchain token ID is different from mainnet token ID.
 to do that, think we should create an util function
```
getMainchainTokenID(ownChainID: Buffer) =>
  Buffer.concat([ownChainID.slice(0, 1), Buffer.alloc(3 + 4, 0)])
```